### ISSUE TITLE
no-implied-eval doesn't consider environments(env)

### ISSUE THREAD

**Issue by yeonjuan** [eslint/eslint#12733]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.13.0
* **npm Version:** v6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
N/A
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

The rule `no-implied-eval` doesn't consider the environment configuration(`env`), while the rule `no-eval` does.

* no-implied-eval - [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8taW1wbGllZC1ldmFsOiBcImVycm9yXCIqL1xuLyplc2xpbnQtZW52IG5vZGUqL1xuXG53aW5kb3cuc2V0VGltZW91dChcInZhciBhID0gMFwiKTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6eyJicm93c2VyIjpmYWxzZX19fQ==)
```js
/*eslint no-implied-eval: "error"*/
/*eslint-env node*/

window.setTimeout("var a = 0"); // 4:1- Implied eval. Consider passing a function instead of a string(no-implied-eval)
```

* no-eval - [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tZXZhbDogXCJlcnJvclwiKi9cbi8qZXNsaW50LWVudiBub2RlKi9cblxud2luZG93LmV2YWwoXCJ2YXIgYSA9IDBcIik7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6eyJqc3giOnRydWV9fSwicnVsZXMiOnt9LCJlbnYiOnsiYnJvd3NlciI6ZmFsc2V9fX0=)
```js
/*eslint no-eval: "error"*/
/*eslint-env node*/

window.eval("var a = 0"); // No Error.
```

**What did you expect to happen?**

No error in `no-implied-eval`, because the environment is configured with 'node'.

**What actually happened? Please include the actual, raw output from ESLint.**
1 Error

**Are you willing to submit a pull request to fix this bug?**
Yes, If it is a bug.

**Comment by kaicataldo** [comment]
Definitely would be nice to have matching behavior here. While we're at it, we could probably also update `no-implied-eval` to account for `globalThis`?
